modules/transactions.js

```diff
@@ -121,6 +121,9 @@ __private.list = function(filter, cb) {
 		maxAmount: '"t_amount" <= ${maxAmount}',
 		type: '"t_type" = ${type}',
 		minConfirmations: 'confirmations >= ${minConfirmations}',
+		data: `ENCODE((SELECT data FROM transfer WHERE transfer."transactionId" = trs_list."t_id"), 'escape') LIKE '${
+			filter.data
+		}'`,
```

Comment:
After research, I suggest for now we go with exact match instead of regexpr: 

1. This must be going to impact the performance 
2. The `encode` with `escape` just escape the special characters so will change the actual content 
   - If user added a transaction with `\u0000 hey how :)`, escaping it will turn it to `\000 hey how :)`
   - If user search with key word `00` the query will match and return the transaction 
3. Since the actual content is different then escaped one so match will not result accurate 

So until we have this `data` field as binary, we should only provide the exact match. Once we find way to convert it to utf8 strings, then we may provide the pattern matching.